admission-webhook/webhook.go

```diff
@@ -392,15 +403,34 @@ func (webhook *webhook) mutateCreateRequest(ctx context.Context, pod *corev1.Pod
 
 	admissionResponse := &admissionV1.AdmissionResponse{Allowed: true}
 
-	if len(patches) != 0 {
-		patchesBytes, err := json.Marshal(patches)
-		if err != nil {
-			return nil, &podAdmissionError{error: fmt.Errorf("unable to marshall patch JSON %v: %v", patches, err), pod: pod, code: http.StatusInternalServerError}
+	if hasGMSA {
+		// pods are GMSA related, we will need further check if we need to randomize the hostname
+		hostName := pod.Spec.Hostname
+		// Patch the hostname only if it is empty
+		if webhook.config.EnableRandomHostName {
+			if hostName == "" {
```

Comment:
I think this could be simplified to avoid to many nested ifs:

```
if hasGMSA && webhook.config.EnableRandomHostName && pod.Spec.Hostname == "" {
 generate addtional patch
}  

if len(patches) != 0 {
  send patches
}
```

Reply:
Will simply, prefer to keep the check of hostname separate to log out the warning information